cli/src/modules/wasm.rs

```diff
@@ -0,0 +1,62 @@
+use super::{Kind, Manager};
+use crate::{
+    spawner::{spawn, SpawnResult},
+    Target, LOCATION,
+};
+use async_trait::async_trait;
+use futures::future::join_all;
+use std::{io, path::PathBuf};
+
+const PATH: &str = "application/apps/rustcore/wasm-bindings";
+
+#[derive(Clone, Debug)]
+pub struct Module {}
+
+impl Module {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl Manager for Module {
+    fn owner(&self) -> Target {
+        Target::Wasm
+    }
+    fn kind(&self) -> Kind {
+        Kind::Rs
+    }
+    fn cwd(&self) -> PathBuf {
+        LOCATION.root.clone().join(PATH)
+    }
+
+    fn deps(&self) -> Vec<Target> {
+        //TODO: Do we have dependencies here?
+        vec![]
+    }
+    fn build_cmd(&self, prod: bool) -> Option<String> {
+        //TODO: It's possible to set the environment with wasm-pack, but it wasn't set in the ruby
+        //version. Check if we can set it here.
+        let env = if prod { "--release" } else { "--dev" };
+
+        Some(format!("wasm-pack build {env} --target bundler"))
+    }
+}
+
+impl Module {
+    // TODO: Use this implementation when testing is implemented in the system
+    #[allow(dead_code)]
+    pub async fn run_tests(&self) -> Vec<Result<SpawnResult, io::Error>> {
+        let path_karma = self.cwd().join("spec");
+
+        join_all([
+            spawn(
+                "wasm-pack test --node",
+                Some(self.cwd()),
+                Some("wasm-pack test wasm-bindings"),
+            ),
+            spawn("npm run test", Some(path_karma), Some("npm test wasm")),
+        ])
+        .await
+    }
```

Comment:
tests one more missed thing. According general concept tests should be a part of `Manager` trait.
- add method `fn build_cmd(&self) -> Option<String>`. If it returns `None` - no tests are required for module. Default value - `None`
- add method `async fn test(&self) -> Result<SpawnResult, Error>`, I think it would be mostly same to method `build(...)`, which already there.

We should not `spawn` anything in the scope of `Module` implementation... all key-functionality like spawning and collecting "feedback" (`stdout`/`stderr`) should happens in same way for all modules and should be implemented in the scope of `trait Module` (like method `build`). In general an idea of `impl Module for Target {}` is: collect inputs/parameters, like paths, commands, maybe some specific arguments and etc.

Indeed, as soon as trait `Module` will get method `test`, it should be reflected in CLI commands `enum Command {...}` (`src/main.rs`)

Reply:
In wasm module we have two commands for the tests (one is for wasm-pack and the other is for npm). Each command has a different cwd to run on. 
I'll change the method `test_cwd()` to let it returns a vector of commands and their corresponding cwds `test_cwd(&self) -> Vec<(PathBuf, String)>`.

With this implementation we won't need the method `fn test_cwd()` anymore